dm/pkg/conn/basedb.go

```diff
@@ -249,7 +259,7 @@ func CloseBaseConnWithoutErr(d *BaseDB, conn *BaseConn) {
 
 // Close release *BaseDB resource.
 func (d *BaseDB) Close() error {
-	if d == nil || d.DB == nil {
+	if d == nil || d.DB == nil || d.doNotClose {
```

Comment:
Why just don't call `Close` in unit tests?

Reply:
In the unit test we need to mock the db operation, if it is closed, go-sql-mock will report the error that database is already closed. So we don't close here, and let the person who writing the test decide when to close

Reply:
Will the `mock.ExpectClose()` help?

Reply:
ExpectClose just mock the close result, the database will still be closed